pystac/link.py

```diff
@@ -334,7 +334,8 @@ def resolve_stac_object(self, root: Optional[Catalog] = None) -> Link:
             and isinstance(self.owner, pystac.Catalog)
         ):
             assert self._target_object
-            self._target_object.set_parent(self.owner)
+            if self._target_object.get_parent() is None:
```

Comment:
PySTAC overrides a parent that we had intentionally set, which is not ideal. But it seems some cases need this...